symengine/cwrapper.cpp

```diff
@@ -489,6 +489,47 @@ char* ascii_art_str()
     return cc;
 }
 
+
+void ntheory_gcd(basic s, const basic a, const basic b)
+{
+    s->m = SymEngine::gcd(static_cast<const Integer &>(*(a->m)), static_cast<const Integer &>(*(b->m)));
```

Comment:
I am not sure if first dereferencing and then casting is the right way to do it when using `RCP`. You might want to use [`rcp_static_cast`](https://github.com/symengine/symengine/blob/master/symengine/symengine_rcp.h#L146) or [`rcp_dynamic_cast`](https://github.com/symengine/symengine/blob/master/symengine/symengine_rcp.h#L154) to first cast the `RCP` and then dereference it.